orchard/orchard_runner.go

```diff
@@ -8,26 +8,89 @@ package orchard
 import (
 	"encoding/json"
 	"fmt"
+	"log"
+	"os"
 	"os/exec"
+	"strings"
+
+	"mce.salesforce.com/sprinkler/common"
 )
 
+const baseDir string = "/sprinkler"
+
 type OrchardRunner interface {
-	Generate(artifact string, command string) string
+	Generate(artifact string, command string) (string, error)
 }
 
 type OrchardStdoutRunner struct{}
 
-func (self OrchardStdoutRunner) Generate(artifact string, command string) (string, error) {
-	cmds, err := parseCommandLine(command)
+func (r OrchardStdoutRunner) Generate(artifact string, command string) (string, error) {
+	if artifact == "" {
+		return processCmd(command, baseDir)
+	}
+
+	if !strings.HasPrefix(artifact, "s3://") {
+		return "", fmt.Errorf("artifact %v is not supported\n", artifact)
+	}
+
+	return s3ArtifactGenerate(artifact, command)
+}
+
+func s3ArtifactGenerate(artifact string, command string) (string, error) {
+
+	// tmp directory to avoid threads race on downloaded artifact
+	tmpDir, err := os.MkdirTemp("", "*")
```

Comment:
how about make it `tmpDir, err := os.MkdirTemp("", "sprinkler-")` instead for easier debugging?

Reply:
tmp dir prefix `sprinkler-` added